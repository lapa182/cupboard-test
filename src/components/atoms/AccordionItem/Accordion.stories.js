import React from "react";

import { storiesOf } from "@storybook/react";
import { withKnobs, text } from "@storybook/addon-knobs";
import AccordionItem from ".";

const accordionTitle = text('Title', 'Red Wine');
const accordionDescription = text('Description', 'Good with red meat');

storiesOf("AccordionItem", module)
    .addDecorator(withKnobs)
    .add("default", () => <AccordionItem title={accordionTitle} description={accordionDescription}/>);
