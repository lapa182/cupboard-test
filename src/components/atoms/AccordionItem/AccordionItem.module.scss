.item {
  margin-bottom: 20px;

  h3 {
    margin-bottom: 10px;
  }

  p {
    font-size: 14px;
    font-style: italic;
  }

  &.active {
    h3 {
      font-weight: bold;
    }
  }
}
