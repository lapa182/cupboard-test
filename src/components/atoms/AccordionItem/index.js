import * as React from "react";
import PropTypes from "prop-types";
import styles from "./AccordionItem.module.scss";

class AccordionItem extends React.Component {
    constructor(props) {
        super(props);
        this.state = {
            active: false
        };
        this._handleClick = this._handleClick.bind(this);
    }

    _handleClick() {
        this.setState({
            active: !this.state.active,
            className: "active"
        });
    }

    render() {
        const activeClass = this.state.active ? styles.active : "";

        return (
            <div className={styles.item + ' ' + activeClass} onClick={this._handleClick}>
                <h3>{this.props.title}</h3>
                <p>{this.props.description}</p>
            </div>
        )
    }
}

AccordionItem.propTypes = {
    title: PropTypes.string.isRequired,
    description: PropTypes.string.isRequired
}

export default AccordionItem;