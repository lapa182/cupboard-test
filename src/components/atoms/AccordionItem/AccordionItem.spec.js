import React from "react";
import renderer from "react-test-renderer";
import AccordionItem from "./index";

describe("Accordion Item", () => {
  it("renders without crashing", () => {
    const accordionItem = renderer
      .create(<AccordionItem title="test" description="test" />)
      .toJSON();
    expect(accordionItem).toMatchSnapshot();
  });
});
