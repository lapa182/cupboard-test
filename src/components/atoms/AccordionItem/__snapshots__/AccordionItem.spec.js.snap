// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Accordion Item renders without crashing 1`] = `
<div
  className="item "
  onClick={[Function]}
  onKeyPress={[Function]}
  role="tab"
  tabIndex="0"
>
  <h3>
    test
  </h3>
  <p>
    test
  </p>
</div>
`;
